public/app/features/scenes/components/layout/SceneGridLayout.tsx

```diff
@@ -0,0 +1,403 @@
+  children: Array<SceneObject<SceneLayoutChildState>>;
+}
+
+export class SceneGridRow extends SceneObjectBase<SceneGridRowState> {
+  static Component = SceneGridRowRenderer;
+  private _originalHeight = 0;
+
+  constructor(
+    state: Omit<SceneGridRowState, 'size'> & { size: Pick<GridCellLayout, 'x' | 'y' | 'height'> & { width?: number } }
+  ) {
+    super({
+      isResizable: true,
+      isDraggable: true,
+      isCollapsible: true,
+      ...state,
+      size: {
+        ...state.size,
+        height: state.isCollapsed ? GRID_CELL_HEIGHT : state.size?.height || DEFAULT_ROW_HEIGHT,
+        width: state.size.width || GRID_COLUMN_COUNT,
+      },
+    });
+
+    this._originalHeight = parseInt(
+      (state.isCollapsed ? GRID_CELL_HEIGHT : state.size?.height || DEFAULT_ROW_HEIGHT).toString(),
+      10
+    );
+
+    this.subs = this.subscribe({
+      next: (state) => {
+        // Preserve the height of the row to be able to restore it when uncollapsing
+        if (
+          state.size &&
+          state.size.height &&
+          state.size.height !== this._originalHeight &&
+          state.size?.height !== GRID_CELL_HEIGHT &&
+          !state.isCollapsed
+        ) {
+          this._originalHeight = parseInt(state.size.height?.toString(), 10);
+        }
+      },
+    });
+  }
+
+  onCollapseToggle = () => {
+    if (!this.state.isCollapsible) {
+      return;
+    }
+    const layout = this.parent;
+
+    if (!layout || !(layout instanceof SceneGridLayout)) {
+      throw new Error('SceneGridRow must be a child of SceneGridLayout');
+    }
+
+    const { isCollapsed, size } = this.state;
+    if (!size) {
+      return;
+    }
+
+    if (layout) {
+      if (isCollapsed) {
+        this.setState({ isCollapsed: false, isResizable: true, size: { ...size, height: this._originalHeight } });
+      } else {
+        this.setState({ isCollapsed: true, isResizable: false, size: { ...size, height: 1 } });
+      }
+      layout.updateLayout();
+    }
+  };
+}
+
+function SceneGridRowRenderer({ model }: SceneComponentProps<SceneGridRow>) {
+  const styles = useStyles2(getSceneGridRowStyles);
+  const { isCollapsible, isCollapsed, title, ...state } = model.useState();
+  const layout = model.getLayout();
+  const isDraggable = layout.state.isDraggable ? state.isDraggable : false;
+  const dragHandle = <SceneDragHandle layoutKey={layout.state.key!} />;
+
+  return (
+    <div className={styles.row}>
+      <div className={cx(styles.rowHeader, isCollapsed && styles.rowHeaderCollapsed)}>
+        <div onClick={model.onCollapseToggle} className={styles.rowTitleWrapper}>
+          {isCollapsible && <Icon name={isCollapsed ? 'angle-right' : 'angle-down'} />}
+          <span className={styles.rowTitle}>{title}</span>
+        </div>
+        {isDraggable && <div>{dragHandle}</div>}
+      </div>
+
+      {!isCollapsed && (
+        <div style={{ display: 'flex', flexGrow: 1, height: 'calc(100%-30px)', width: '100%' }}>
```

Comment:
don't think this 30px is correct (the row header has margin bottom 8px as well). And it's just better to use flex instead calc, but we can fix this later